### PR TITLE
Add Metroid Goal to Fusion

### PR DESCRIPTION
**Pull request by JeffGainsNGames**
Adds the goal of collecting infant/baby metroids. Configurable on bosses or anywhere. Add associated tests.

Also adds basic testing for DB.

### CODE REVIEW

**Comment by Miepee** [comment]
test

**Review by dyceron — CHANGES_REQUESTED** [review]
mypy errors. Also add a screenshot of the goal tab to the comment

File: randovania/games/fusion/logic_database/Sector 3 PYR.txt [hunk 1]
```diff
@@ -143,7 +143,7 @@ Extra - room_id: [3]
       Any of the following:
           Wall Jump (Beginner) or Can Climb High Jump Wall
           Speed Booster and Shinespark Tricks (Beginner)
-          After Level 2 Locks Released and Stand On Frozen Enemies (Beginner)
+          After Level 2 Locks Released and Stand On Frozen Enemies (Beginner) and Can Freeze Enemies
```
Comment (dyceron) [inline randovania/games/fusion/logic_database/Sector 3 PYR.txt:146]: I assume this wasn't meant for this PR, but oh well

Comment (JeffGainsNGames) [reply randovania/games/fusion/logic_database/Sector 3 PYR.txt:146]: It was a part of `test_fusion.py`.

**Comment by JeffGainsNGames** [comment]
image
There may be additional restricted placement settings, so left it as a radio button + checkbox combo.

File: randovania/gui/ui_files/preset_fusion_goal.ui [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+           </property>
+           <property name="wordWrap">
+            <bool>true</bool>
+           </property>
+          </widget>
+         </item>
+         <item>
+          <widget class="QCheckBox" name="prefer_bosses_check">
+           <property name="text">
+            <string>Prefer Bosses (Arachnus, Charge Core-X, Zazabi, etc)</string>
```
Comment (Miepee) [inline randovania/gui/ui_files/preset_fusion_goal.ui:119]: would list out all bosses here than just sying etc

File: test/games/fusion/test_fusion.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    rl = game.region_list
+    db = game.resource_database
+    expected_dict: dict[Any, Any] = {}
+    database_dict: dict[Any, Any] = {}
+    for i in range(len(resource_list)):
+        database_dict[resource_list[i]] = []
+        expected_dict[resource_list[i]] = []
+
+    for node in rl.all_nodes:
+        for dest_node, requirements in rl.area_connections_from(node):
```
Comment (Miepee) [inline test/games/fusion/test_fusion.py:49]: can you try whether an `assert node is not None` here gets rid of the errors below?

Comment (JeffGainsNGames) [reply test/games/fusion/test_fusion.py:49]: it does, the above mypy error is still present.

**Review by Miepee — APPROVED** [review]

**Pull request merged**